File: VB/MVCSchedulerReadOnly/Models/SchedulerDataObject.vb
Imports System.Linq

Namespace MVCSchedulerReadOnly.Models
    #Region "#SchedulerDataObject"
    Public Class SchedulerDataObject
        Public Property Appointments() As System.Collections.IEnumerable
        Public Property Resources() As System.Collections.IEnumerable
    End Class
    #End Region ' #SchedulerDataObject

    #Region "#SchedulerDataHelper"
    Public Class SchedulerDataHelper
        Public Shared Function GetResources() As System.Collections.IEnumerable
            Dim db As New SchedulingDataClassesDataContext()
            Return From res In db.DBResources _
                   Select res
        End Function
        Public Shared Function GetAppointments() As System.Collections.IEnumerable
            Dim db As New SchedulingDataClassesDataContext()
            Return From apt In db.DBAppointments _
                   Select apt
        End Function
        Public Shared ReadOnly Property DataObject() As SchedulerDataObject
            Get
                Return New SchedulerDataObject() With {.Appointments = GetAppointments(), .Resources = GetResources()}
            End Get
        End Property
    End Class
    #End Region ' #SchedulerDataHelper
End Namespace

File: CS/MVCSchedulerReadOnly/Models/SchedulerDataObject.cs
using System.Linq;

namespace MVCSchedulerReadOnly.Models
{
    #region #SchedulerDataObject
    public class SchedulerDataObject
    {
        public System.Collections.IEnumerable Appointments { get; set; }
        public System.Collections.IEnumerable Resources { get; set; }
    }
    #endregion #SchedulerDataObject

    #region #SchedulerDataHelper
    public class SchedulerDataHelper
    {
        public static System.Collections.IEnumerable GetResources()
        {
            SchedulingDataClassesDataContext db = new SchedulingDataClassesDataContext();
            return from res in db.DBResources select res;
        }
        public static System.Collections.IEnumerable GetAppointments()
        {
            SchedulingDataClassesDataContext db = new SchedulingDataClassesDataContext();
            return from apt in db.DBAppointments select apt;
        }
        public static SchedulerDataObject DataObject
        {
            get
            {
                return new SchedulerDataObject()
                {
                    Appointments = GetAppointments(),
                    Resources = GetResources()
                };
            }
        }
    }
    #endregion #SchedulerDataHelper
}